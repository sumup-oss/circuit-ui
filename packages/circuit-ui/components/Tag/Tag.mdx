import { Meta, Status, Props, Story } from '../../../../.storybook/components';
import * as Stories from './Tag.stories';

<Meta of={Stories} />

# Tag

<Status.Stable />

Tags enable the user to add or remove attributes to a given content, for
example: transaction history for a given day or week.

<Story of={Stories.Base} />
<Props />

## Best practices

- For content within the tag, the text should be no longer than 30 characters.
- Be as concise as possible in wording, limit to one to two words, as to not
  take up too much space.

## Usage guidelines

- **Do** make tags removable by default in order for users to have the
  ability to add and remove them.
- **Do** display tags above or within proximity to the content that needs to be
  filtered.
- **Do** ensure that icons are always displayed on the left of the label, in
  a canvas of 24px.

## Component variations

### Selected

<Story of={Stories.Selected} />

### With prefix

<Story of={Stories.WithPrefix} />

### With suffix

<Story of={Stories.WithSuffix} />

### Clickable

<Story of={Stories.Clickable} />

### Removable

<Story of={Stories.Removable} />
