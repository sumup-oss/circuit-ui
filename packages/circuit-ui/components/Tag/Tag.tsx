/**
 * Copyright 2019, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/** @jsx jsx */
import React, { HTMLProps, Ref, FC, SVGProps, MouseEvent } from 'react';
import { css, jsx } from '@emotion/core';
import { Dispatch as TrackingProps } from '@sumup/collector';
import { Theme } from '@sumup/design-tokens';

import styled, { StyleProps } from '../../styles/styled';
import { textMega, focusOutline } from '../../styles/style-mixins';
import useClickHandler from '../../hooks/use-click-handler';
import { CloseButton, CloseButtonProps } from '../CloseButton/CloseButton';

interface BaseProps {
  /**
   * Render prop that should render a leading-aligned icon or element.
   */
  prefix?: FC<SVGProps<SVGSVGElement>>;
  /**
   * Render prop that should render a trailing-aligned icon or element.
   */
  suffix?: FC<SVGProps<SVGSVGElement>>;
  /**
   * Renders a close button inside the tag and calls the provided function
   * when the button is clicked.
   */
  onRemove?: (event: MouseEvent) => void;
  /**
   * Text label for the remove icon for screen readers.
   * Important for accessibility.
   */
  labelRemoveButton?: string;
  /**
   * Triggers selected styles on the tag.
   */
  selected?: boolean;
  /**
   * Additional data that is dispatched with the tracking event.
   */
  tracking?: TrackingProps;
  /**
   *  The ref to the DOM element
   */
  ref?: Ref<HTMLDivElement & HTMLButtonElement>;
}

type DivElProps = Omit<HTMLProps<HTMLDivElement>, 'prefix'>;
type ButtonElProps = Omit<HTMLProps<HTMLButtonElement>, 'prefix'>;

export type TagProps = BaseProps & DivElProps & ButtonElProps;

const BORDER_WIDTH = '1px';

type TagElProps = Omit<TagProps, 'prefix' | 'suffix' | 'labelRemoveButton'> & {
  removable: boolean;
};

const tagBaseStyles = ({ theme }: StyleProps) => css`
  label: tag;
  display: inline-flex;
  align-items: center;
  margin: 0;
  ${textMega({ theme })};
  word-break: break-word;
  border: ${BORDER_WIDTH} solid ${theme.colors.n300};
  border-radius: ${theme.borderRadius.giga};
  padding: ${theme.spacings.bit} ${theme.spacings.kilo};
  cursor: default;
  background-color: ${theme.colors.white};
  transition: opacity ${theme.transitions.default},
    color ${theme.transitions.default},
    background-color ${theme.transitions.default},
    border-color ${theme.transitions.default};
`;

const tagRemovableStyles = ({ theme, removable }: StyleProps & TagElProps) =>
  removable &&
  css`
    label: tag--removable;
    padding-right: calc(${theme.spacings.bit} + ${theme.spacings.tera});
  `;

const tagClickableStyles = ({ theme, onClick }: StyleProps & TagElProps) =>
  onClick &&
  css`
    label: tag--clickable;
    cursor: pointer;
    outline: 0;
    text-align: left;

    &:active {
      color: ${theme.colors.bodyColor};
    }

    &:hover {
      background-color: ${theme.colors.n200};
      border-color: ${theme.colors.n500};
    }

    &:focus {
      ${focusOutline({ theme })};
    }
  `;

const tagSelectedStyles = ({ theme, selected }: StyleProps & TagElProps) =>
  selected &&
  css`
    label: tag--selected;
    background-color: ${theme.colors.p500};
    border-color: ${theme.colors.p700};
    color: ${theme.colors.white};
  `;

const tagSelectedClickableStyles = ({
  theme,
  selected,
  onClick,
}: StyleProps & TagElProps) =>
  selected &&
  onClick &&
  css`
    label: tag--selected--clickable;

    &:active {
      color: ${theme.colors.white};
    }

    &:hover {
      background-color: ${theme.colors.p700};
      border-color: ${theme.colors.p700};
    }
  `;

const TagElement = styled('div')<TagElProps>(
  tagBaseStyles,
  tagRemovableStyles,
  tagClickableStyles,
  tagSelectedStyles,
  tagSelectedClickableStyles,
);

const prefixStyles = (theme: Theme) => css`
  label: tag__prefix;
  flex-shrink: 0;
  margin-left: -${theme.spacings.bit};
  margin-right: ${theme.spacings.bit};
`;

const suffixStyles = (theme: Theme) => css`
  label: tag__suffix;
  flex-shrink: 0;
  margin-left: ${theme.spacings.bit};
  margin-right: -${theme.spacings.bit};
`;

const closeButtonStyles = ({ theme }: StyleProps) => css`
  label: tag__close-button;
  position: absolute;
  top: 50%;
  right: ${BORDER_WIDTH};
  transform: translateY(-50%);
  border-radius: ${theme.borderRadius.mega};
`;

const RemoveButton = styled(CloseButton)<CloseButtonProps>(closeButtonStyles);

const Container = styled.div`
  label: tag__container;
  position: relative;
`;

export const Tag = React.forwardRef(
  (
    {
      children,
      prefix: Prefix,
      suffix: Suffix,
      onRemove,
      labelRemoveButton = 'Remove',
      selected,
      onClick,
      tracking,
      className,
      style,
      ...props
    }: TagProps,
    ref: BaseProps['ref'],
  ) => {
    const as = onClick ? 'button' : 'div';
    const handleClick = useClickHandler<MouseEvent<any>>(
      onClick,
      tracking,
      'tag',
    );
    const removable = Boolean(onRemove);

    return (
      <Container className={className} style={style}>
        <TagElement
          removable={removable}
          selected={selected}
          onClick={handleClick}
          type={onClick && 'button'}
          as={as}
          ref={ref}
          {...props}
        >
          {Prefix && <Prefix css={prefixStyles} />}

          {children}

          {Suffix && <Suffix css={suffixStyles} />}
        </TagElement>

        {removable && (
          <RemoveButton
            type="button"
            variant={selected ? 'primary' : 'secondary'}
            selected={selected}
            label={labelRemoveButton}
            data-testid="tag-close"
            size="kilo"
            onClick={onRemove}
            tracking={{
              component: 'tag-remove',
              ...tracking,
            }}
          />
        )}
      </Container>
    );
  },
);

Tag.displayName = 'Tag';
