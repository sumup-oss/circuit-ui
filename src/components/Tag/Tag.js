/**
 * Copyright 2019, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import { css } from '@emotion/core';

import {
  eitherOrPropType,
  childrenPropType
} from '../../util/shared-prop-types';
import { textMega, shadowBorder } from '../../styles/style-helpers';
import DefaultCloseButton from '../CloseButton';

const closeButtonStyles = ({ theme }) => css`
  label: tag__close-button;
  margin-left: ${theme.spacings.kilo};
  vertical-align: middle;
`;

const closeButtonSelectedStyles = ({ selected, theme }) =>
  selected &&
  css`
    label: tag__close-button--selected;

    > svg {
      fill: ${theme.colors.white};
      vertical-align: inherit;
    }
  `;

const CloseButton = styled(DefaultCloseButton)(
  closeButtonStyles,
  closeButtonSelectedStyles
);

const tagStyles = ({ theme }) => css`
  label: tag;
  border-radius: ${theme.borderRadius.mega};

  ${textMega({ theme })};
  ${shadowBorder(theme.colors.n300)};
  padding: ${theme.spacings.bit} ${theme.spacings.kilo};
  cursor: default;
  display: inline-block;
`;

const tagClickableStyles = ({ onClick, theme }) =>
  onClick &&
  css`
    label: tag--clickable;
    cursor: pointer;

    &:hover {
      background-color: ${theme.colors.n300};
      ${shadowBorder(theme.colors.n300)};
    }
  `;

const tagSelectedStyles = ({ selected, theme }) =>
  selected &&
  css`
    label: tag--selected;
    background-color: ${theme.colors.p500};
    ${shadowBorder(theme.colors.p500)};
    color: ${theme.colors.white};
  `;

const tagSelectedClickableStyles = ({ selected, onClick, theme }) =>
  selected &&
  onClick &&
  css`
    &:hover {
      background-color: ${theme.colors.p500};
      ${shadowBorder(theme.colors.p500)};
    }
  `;

const iconStyles = ({ theme }) => css`
  label: tag__icon;
  margin-right: ${theme.spacings.bit};
  display: inline-block;
  width: ${theme.spacings.mega};
  height: ${theme.spacings.mega};
  vertical-align: middle;

  > svg {
    vertical-align: top;
  }
`;

const iconSelectedStyles = ({ selected, theme }) =>
  selected &&
  css`
    label: tag__icon--selected;

    > svg {
      fill: ${theme.colors.white};
    }
  `;

const IconContainer = styled('span')`
  ${iconStyles};
  ${iconSelectedStyles};
`;
const TagElement = styled('span')`
  ${tagStyles};
  ${tagClickableStyles};
  ${tagSelectedStyles};
  ${tagSelectedClickableStyles};
`;

/**
 * Tag component
 */
const Tag = ({
  children,
  icon,
  onRemove,
  labelRemoveButton,
  selected,
  ...props
}) => (
  <TagElement {...{ selected, ...props }}>
    {!onRemove && icon && (
      <IconContainer {...{ selected }}>{icon}</IconContainer>
    )}
    {children}
    {onRemove && (
      <CloseButton
        onClick={onRemove}
        selected={selected}
        label={labelRemoveButton}
        data-testid="tag-close"
      />
    )}
  </TagElement>
);

Tag.propTypes = {
  /**
   * The content of the tag.
   */
  children: childrenPropType,
  /**
   * An optional  tag's icon.
   */
  icon: eitherOrPropType('icon', 'onRemove', PropTypes.element),
  /**
   * Renders a close button inside the tag and calls the provided function
   * when the button is clicked.
   */
  onRemove: eitherOrPropType('icon', 'onRemove', PropTypes.func),
  /**
   * Text label for the remove icon for screen readers.
   * Important for accessibility.
   */
  labelRemoveButton: PropTypes.string,
  /**
   * Triggers selected styles on the tag.
   */
  selected: PropTypes.bool
};

Tag.defaultProps = {
  children: null,
  icon: null,
  onRemove: null,
  selected: false,
  labelRemoveButton: 'remove'
};

/**
 * @component
 */
export default Tag;
