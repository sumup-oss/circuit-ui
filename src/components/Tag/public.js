export { default as Tag } from '.';
