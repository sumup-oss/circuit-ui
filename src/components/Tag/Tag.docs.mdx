import { Status, Props, Story } from '../../../.storybook/components';
import Tag from './Tag';

# Tag

<Status.Stable />

Tags enable the user to add or remove attributes to a given content, for
example: transaction history for a given day or week.

<Story id="components-tag--base" />

<Props of={Tag} />

## Best practices

- For content within the tag, the text should be no longer than 30 characters.
- Be as concise as possible in wording, limit to one to two words, as to not
  take up too much space.

## Usage guidelines

- **Do** make tags removable by default in order for users to have the
  ability to add and remove them.
- **Do** display tags above or within proximity to the content that needs to be
  filtered.
- **Do** ensure that icons are always displayed on the left of the label, in
  a canvas of 24px.

## Component variations

### Selected

<Story id="components-tag--selected" />

### With icon

<Story id="components-tag--with-icon" />

### Clickable

<Story id="components-tag--clickable" />

### Removable

<Story id="components-tag--removable" />
