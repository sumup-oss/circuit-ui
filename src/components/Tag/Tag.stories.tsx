/**
 * Copyright 2019, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import { action } from '@storybook/addon-actions';
import { Check } from '@sumup/icons';

import docs from './Tag.docs.mdx';
import { Tag, TagProps } from './Tag';

export default {
  title: 'Components/Tag',
  component: Tag,
  parameters: {
    docs: { page: docs },
  },
};
export const Base = ({ onRemove, ...args }: TagProps) => (
  <Tag {...args}>Transactions</Tag>
);

export const Selected = ({ onRemove, ...args }: TagProps) => (
  <Tag {...args}>Transactions</Tag>
);

Selected.args = { selected: true };

export const WithPrefix = ({ onRemove, ...args }: TagProps) => (
  <Tag {...args}>Transactions</Tag>
);

WithPrefix.args = { prefix: Check };

export const WithSuffix = ({ onRemove, ...args }: TagProps) => (
  <Tag {...args}>Transactions</Tag>
);

WithSuffix.args = { suffix: Check };

export const Clickable = ({ onRemove, ...args }: TagProps) => (
  <Tag {...args}>Transactions</Tag>
);

Clickable.args = {
  onClick: action('Tag clicked'),
  tracking: { label: 'trackingId' },
};

export const Removable = (args: TagProps) => <Tag {...args}>Transactions</Tag>;

Removable.args = {
  onRemove: action('Tag removed'),
  labelRemoveButton: 'Remove',
  tracking: { label: 'trackingId' },
};
