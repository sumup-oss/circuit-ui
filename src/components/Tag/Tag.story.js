/**
 * Copyright 2019, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import { action } from '@storybook/addon-actions';
import { boolean } from '@storybook/addon-knobs/react';

import docs from './Tag.docs.mdx';
import Tag from './Tag';

const Icon = () => (
  <svg
    fill="#000000"
    height="16"
    viewBox="0 0 24 24"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path d="M0 0h24v24H0z" fill="none" />
    <path d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z" />
  </svg>
);

export default {
  title: 'Components|Tag',
  component: Tag,
  parameters: {
    docs: { page: docs },
    jest: ['Tag']
  }
};

export const base = () => (
  <Tag
    selected={boolean('Selected', false)}
    onRemove={boolean('Removable', false) ? action('Tag removed') : null}
    icon={boolean('With Icon', false) ? <Icon /> : null}
    onClick={boolean('Clickable', false) ? action('Tag clicked') : null}
  >
    Transactions
  </Tag>
);

export const selected = () => <Tag selected>Transactions</Tag>;

export const withIcon = () => <Tag icon={<Icon />}>Transactions</Tag>;

export const removable = () => (
  <Tag onRemove={action('Tag removed')} labelRemoveButton="Remove">
    Transactions
  </Tag>
);

export const clickable = () => (
  <Tag onClick={action('Tag clicked')}>Transactions</Tag>
);
