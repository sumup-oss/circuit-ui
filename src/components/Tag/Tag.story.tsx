/**
 * Copyright 2019, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import { action } from '@storybook/addon-actions';
import { boolean, text } from '@storybook/addon-knobs';
import { Check } from '@sumup/icons';

import docs from './Tag.docs.mdx';
import { Tag, TagProps } from './Tag';

export default {
  title: 'Components/Tag',
  component: Tag,
  parameters: {
    docs: { page: docs },
  },
};

const BaseTag = (props: Partial<TagProps>) => (
  <Tag
    selected={boolean('Selected', false)}
    prefix={boolean('Prefix', false) ? Check : undefined}
    suffix={boolean('Suffix', false) ? Check : undefined}
    onClick={boolean('Clickable', false) ? action('Tag clicked') : undefined}
    onRemove={boolean('Removable', false) ? action('Tag removed') : undefined}
    {...props}
  >
    Transactions
  </Tag>
);

export const base = () => <BaseTag />;

export const selected = () => <BaseTag selected />;

export const withPrefix = () => <BaseTag prefix={Check} />;

export const withSuffix = () => <BaseTag suffix={Check} />;

export const clickable = () => (
  <BaseTag
    onClick={action('Tag clicked')}
    as="button"
    tracking={{ label: text('Tracking Label', 'trackingId') }}
  />
);

export const removable = () => (
  <BaseTag
    onRemove={action('Tag removed')}
    labelRemoveButton="Remove"
    tracking={{ label: text('Tracking Label', 'trackingId') }}
  />
);
