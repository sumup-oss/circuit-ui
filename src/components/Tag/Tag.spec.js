/**
 * Copyright 2019, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';

import Tag from '.';

const DummyIcon = props => <div data-testid="tag-icon" {...props} />;

describe('Tag', () => {
  /**
   * Style tests.
   */
  describe('when is default', () => {
    const props = {};

    it('should render with default styles', () => {
      const component = create(<Tag {...props}>SomeTest</Tag>);
      expect(component).toMatchSnapshot();
    });
  });

  describe('when is clickable', () => {
    const props = {
      onClick: jest.fn()
    };

    it('should render with clickable styles', () => {
      const component = create(<Tag {...props}>SomeTest</Tag>);
      expect(component).toMatchSnapshot();
    });
  });

  describe('when is selected', () => {
    const props = {
      selected: true
    };

    it('should render with selected styles', () => {
      const component = create(<Tag {...props}>SomeTest</Tag>);
      expect(component).toMatchSnapshot();
    });

    it('should change the given icon color', () => {
      const component = create(
        <Tag {...{ prefix: DummyIcon, ...props }}>SomeTest</Tag>
      );
      expect(component).toMatchSnapshot();
    });

    it('should change the close icon color', () => {
      const onRemove = jest.fn();

      const component = create(<Tag {...{ onRemove, ...props }}>SomeTest</Tag>);

      expect(component).toMatchSnapshot();
    });
  });

  /**
   * Accessibility tests.
   */
  it('should meet accessibility guidelines', async () => {
    const wrapper = renderToHtml(<Tag>Tag</Tag>);
    const actual = await axe(wrapper);
    expect(actual).toHaveNoViolations();
  });

  /**
   * Logic tests.
   */
  describe('when is removable', () => {
    const props = {
      onRemove: jest.fn()
    };

    it('should render a close button', () => {
      const { getByTestId } = render(<Tag {...props}>SomeTest</Tag>);
      expect(getByTestId('tag-close')).not.toBeNull();
    });

    it('should calls onRemove when click close', () => {
      const { getByTestId } = render(<Tag {...props}>SomeTest</Tag>);

      act(() => {
        fireEvent.click(getByTestId('tag-close'));
      });

      expect(props.onRemove).toHaveBeenCalledTimes(1);
    });
  });

  describe('when a suffix prop is passed', () => {
    const props = {
      suffix: DummyIcon
    };

    it('should render with suffix', () => {
      const { getByTestId } = render(<Tag {...props}>SomeTest</Tag>);
      expect(getByTestId('tag-icon')).not.toBeNull();
    });

    it('gives priority to close button when removable', () => {
      const onRemove = jest.fn();

      const { queryByTestId } = render(
        <Tag {...{ onRemove, props }}>SomeTest</Tag>
      );

      expect(queryByTestId('tag-icon')).toBeNull();
      expect(queryByTestId('tag-close')).not.toBeNull();
    });
  });

  describe('when a prefix prop is passed', () => {
    const props = {
      prefix: DummyIcon
    };

    it('should render with a prefix', () => {
      const { getByTestId } = render(<Tag {...props}>SomeTest</Tag>);
      expect(getByTestId('tag-icon')).not.toBeNull();
    });
  });
});
