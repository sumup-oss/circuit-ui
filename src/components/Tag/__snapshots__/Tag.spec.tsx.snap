// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag when is clickable should render with clickable styles 1`] = `
.circuit-1 {
  position: relative;
}

.circuit-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 16px;
  line-height: 24px;
  border: 1px solid #CCC;
  border-radius: 6px;
  padding: 4px 12px;
  cursor: default;
  background-color: #FFF;
  -webkit-transition: opacity 120ms ease-in-out, color 120ms ease-in-out, background-color 120ms ease-in-out, border-color 120ms ease-in-out;
  transition: opacity 120ms ease-in-out, color 120ms ease-in-out, background-color 120ms ease-in-out, border-color 120ms ease-in-out;
  cursor: pointer;
  outline: 0;
}

.circuit-0:active {
  color: #0F131A;
}

.circuit-0:hover {
  background-color: #E6E6E6;
  border-color: #999;
}

.circuit-0:focus {
  outline: 0;
  box-shadow: 0 0 0 4px #AFD0FE;
}

.circuit-0:focus::-moz-focus-inner {
  border: 0;
}

<div
  class="circuit-1"
>
  <button
    class="circuit-0"
    type="button"
  >
    SomeTest
  </button>
</div>
`;

exports[`Tag when is default should render with default styles 1`] = `
.circuit-1 {
  position: relative;
}

.circuit-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 16px;
  line-height: 24px;
  border: 1px solid #CCC;
  border-radius: 6px;
  padding: 4px 12px;
  cursor: default;
  background-color: #FFF;
  -webkit-transition: opacity 120ms ease-in-out, color 120ms ease-in-out, background-color 120ms ease-in-out, border-color 120ms ease-in-out;
  transition: opacity 120ms ease-in-out, color 120ms ease-in-out, background-color 120ms ease-in-out, border-color 120ms ease-in-out;
}

<div
  class="circuit-1"
>
  <div
    class="circuit-0"
  >
    SomeTest
  </div>
</div>
`;

exports[`Tag when is selected should change the close icon color 1`] = `
.circuit-3 {
  position: relative;
}

.circuit-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 16px;
  line-height: 24px;
  border: 1px solid #CCC;
  border-radius: 6px;
  padding: 4px 12px;
  cursor: default;
  background-color: #FFF;
  -webkit-transition: opacity 120ms ease-in-out, color 120ms ease-in-out, background-color 120ms ease-in-out, border-color 120ms ease-in-out;
  transition: opacity 120ms ease-in-out, color 120ms ease-in-out, background-color 120ms ease-in-out, border-color 120ms ease-in-out;
  padding-right: calc(4px + 32px);
  background-color: #3063E9;
  border-color: #234BC3;
  color: #FFF;
}

.circuit-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: auto;
  height: auto;
  margin: 0;
  cursor: pointer;
  font-size: 16px;
  line-height: 24px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 700;
  border-width: 1px;
  border-style: solid;
  -webkit-transition: opacity 120ms ease-in-out, color 120ms ease-in-out, background-color 120ms ease-in-out, border-color 120ms ease-in-out;
  transition: opacity 120ms ease-in-out, color 120ms ease-in-out, background-color 120ms ease-in-out, border-color 120ms ease-in-out;
  background-color: #3063E9;
  border-color: #3063E9;
  color: #FFF;
  padding: 8px calc(24px - 1px);
  border-radius: 8px;
  padding: 8px;
  border: 0;
  position: absolute;
  top: 1px;
  right: 1px;
  border-radius: 4px;
}

.circuit-2:focus {
  outline: 0;
  box-shadow: 0 0 0 4px #AFD0FE;
}

.circuit-2:focus::-moz-focus-inner {
  border: 0;
}

.circuit-2:disabled,
.circuit-2[disabled] {
  opacity: 0.5;
  pointer-events: none;
  box-shadow: none;
}

.circuit-2:hover {
  background-color: #234BC3;
  border-color: #234BC3;
}

.circuit-2:active {
  background-color: #1A368E;
  border-color: #1A368E;
}

.circuit-1 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

<div
  class="circuit-3"
>
  <div
    class="circuit-0"
  >
    SomeTest
  </div>
  <button
    class="circuit-2"
    data-testid="tag-close"
    title="Remove"
    type="button"
  >
    <svg
      fill="none"
      height="16"
      role="presentation"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3 3l10 10m0-10L3 13"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-width="2"
      />
    </svg>
    <span
      class="circuit-1"
    >
      Remove
    </span>
  </button>
</div>
`;

exports[`Tag when is selected should change the given icon color 1`] = `
.circuit-2 {
  position: relative;
}

.circuit-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 16px;
  line-height: 24px;
  border: 1px solid #CCC;
  border-radius: 6px;
  padding: 4px 12px;
  cursor: default;
  background-color: #FFF;
  -webkit-transition: opacity 120ms ease-in-out, color 120ms ease-in-out, background-color 120ms ease-in-out, border-color 120ms ease-in-out;
  transition: opacity 120ms ease-in-out, color 120ms ease-in-out, background-color 120ms ease-in-out, border-color 120ms ease-in-out;
  background-color: #3063E9;
  border-color: #234BC3;
  color: #FFF;
}

.circuit-0 {
  margin-left: -4px;
  margin-right: 4px;
}

<div
  class="circuit-2"
>
  <div
    class="circuit-1"
  >
    <div
      class="circuit-0"
      data-testid="tag-icon"
    />
    SomeTest
  </div>
</div>
`;

exports[`Tag when is selected should render with selected styles 1`] = `
.circuit-1 {
  position: relative;
}

.circuit-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 16px;
  line-height: 24px;
  border: 1px solid #CCC;
  border-radius: 6px;
  padding: 4px 12px;
  cursor: default;
  background-color: #FFF;
  -webkit-transition: opacity 120ms ease-in-out, color 120ms ease-in-out, background-color 120ms ease-in-out, border-color 120ms ease-in-out;
  transition: opacity 120ms ease-in-out, color 120ms ease-in-out, background-color 120ms ease-in-out, border-color 120ms ease-in-out;
  background-color: #3063E9;
  border-color: #234BC3;
  color: #FFF;
}

<div
  class="circuit-1"
>
  <div
    class="circuit-0"
  >
    SomeTest
  </div>
</div>
`;
